### DISCUSSION TITLE
How to save page source

### DISCUSSION THREAD

**Discussion by bidak** [php-webdriver/php-webdriver#512] [Q&A] [answered]
Can i save a web page with this driver?
I'm trying to save this by ctrl + s but my script didn't work.
```
$this->webDriver->action()->keyDown(null, WebDriverKeys::CONTROL)->sendKeys(null, 's');
```
and:
```
$this->webDriver->getKeyboard()->pressKey(WebDriverKeys::LEFT_CONTROL)->sendKeys('s');
```
what's the solution?

* Php-webdriver version: latest version
* PHP version: 5.6.31
* Selenium server version: 3.7.1
* Operating system: windows
* Browser used + version: chrome v62

**Comment by paulbriton** [comment, marked as answer]
AFAIK, interacting with OS dialog box such as print dialog and save dialog (like the one opened with ctrl + s) is a real pain with selenium. 

As a workaround, you can call `$driver->getPageSource();` and save the result into a file.
